core/bloombits/scheduler.go

```diff
@@ -150,6 +150,7 @@ func (s *scheduler) scheduleDeliveries(pend chan uint64, done chan []byte, quit
 			// Wait until the request is honoured
 			s.lock.Lock()
 			res := s.responses[idx]
+			cached := res.cached
```

Comment:
This is not correct because res.cached will be nil until res.done is closed.

Reply:
Yes, we are working on a tool. We still have about a score of bugs to report.

Reply:
I just wonder if `res.cached` should be protected by s.lock.Lock() here.

Reply:
It cannot be protected like that. The idea in this code is that `s.responses` keeps track of all the response objects. When a request is fulfilled, `resp.cache` is set to the response value and then `resp.done` is closed, unblocking the select statement here.

Reply:
Thank you for your elaborate explanation. I will close this PR. :)